lxd/certificates.go

```diff
@@ -354,6 +354,88 @@ func clusterMemberJoinTokenValid(d *Daemon, r *http.Request, projectName string,
 	return nil, nil
 }
 
+// certificateTokenValid searches for certificate token that matches the joint token provided.
+// Returns matching operation if found and cancels the operation, otherwise returns nil.
+func certificateTokenValid(d *Daemon, r *http.Request, projectName string, joinToken *api.CertificateAddToken) (*api.Operation, error) {
```

Comment:
So this feels wrong.

What I'm expecting is:
 - Admin issues a token through `POST /1.0/certificates`
 - The `req` (CertificatesPost) data is stored in a metadata field of a token operation
 - The admin is returned the token which includes the address of the server, fingerprint and secret

 - The client uses the token, connects to the server at the address provided, validates the fingerprint using the one provided, then adds itself to the trust store by doing `POST /1.0/certificates` with the token
 - The server locates the operation for that token, retrieves the initial `req` from the operation and adds the client's certificate to the trust store using that recorded `req`

That means the user when provided a token will not necessarily know what kind of restrictions the administrator may have put in place on it and that's kinda the point as providing that information may cause a data leak of some kind.